src/appleseed/renderer/modeling/environmentshader/edfenvironmentshader.cpp

```diff
@@ -97,15 +98,17 @@ namespace
             if (!EnvironmentShader::on_frame_begin(project, parent, recorder, abort_switch))
                 return false;
 
+            const OnFrameBeginMessageContext context("environment edf", this);
+
             // Cache the bound environment EDF.
             m_env_edf = dynamic_cast<const EnvironmentEDF*>(m_inputs.get_entity("environment_edf"));
 
             // Abort if no environment EDF was found.
             if (m_env_edf == nullptr)
             {
                 RENDERER_LOG_ERROR(
-                    "while preparing environment shader \"%s\": "
-                    "cannot find environment edf \"%s\".",
+                    "%scannot find environment edf \"%s\".",
```

Comment:
Missing space

Reply:
On purpose. The first `%s` is either an empty string or a prefix such as `while defining entity X: ` (with a trailing space).